pragma-node/src/main.rs

```diff
@@ -105,9 +112,16 @@ async fn main() {
         pragma_entities::connection::init_pool("pragma-node-api", ENV_POSTGRES_DATABASE_URL)
             .expect("can't init postgres (onchain db) pool");
 
+    let pragma_signer = if config.is_production_mode() {
```

Comment:
we could use a builder pattern here for simplicity and to make it easy to add different signer modes in the future. Let's create the issue and tackle it later